formats/src/main/scala/overflowdb/formats/neo4jcsv/Neo4jCsvImporter.scala

```diff
@@ -0,0 +1,244 @@
+    }
+  }
+
+  private def parseNodeRowData(columnsRaw: Seq[String], lineNo: Int, columnDefs: Map[Int, ColumnDef]): ParsedNodeRowData = {
+    var id: Integer = null
+    var label: String = null
+    val properties = Seq.newBuilder[ParsedProperty]
+    columnsRaw.zipWithIndex.foreach { case (entry, idx) =>
+      assert(columnDefs.contains(idx), s"column with index=$idx not found in column definitions derived from headerFile")
+      columnDefs(idx) match {
+        case ColumnDef(_, ColumnType.Id, _) =>
+          id = entry.toInt
+        case ColumnDef(_, ColumnType.Label, _) =>
+          label = entry
+        case ColumnDef(Some(name), valueType, false) =>
+          parseProperty(entry, name, valueType).foreach(properties.addOne)
+        case ColumnDef(Some(name), valueType, true) =>
+          parseArrayProperty(entry, name, valueType).foreach(properties.addOne)
+        case other =>
+          throw new MatchError(s"unhandled case $other")
+      }
+    }
+    assert(id != null, s"no ID column found in line $lineNo")
+    assert(label != null, s"no LABEL column found in line $lineNo")
+
+    val ret = ParsedNodeRowData(id, label, properties.result())
+    logger.debug("parsed line {}: {}", lineNo, ret)
+    ret
+  }
+
+  private def parseEdgeRowData(columnsRaw: Seq[String], lineNo: Int, columnDefs: Map[Int, ColumnDef]): ParsedEdgeRowData = {
+    var startId: Integer = null
+    var endId: Integer = null
+    var label: String = null
+    val properties = Seq.newBuilder[ParsedProperty]
+    columnsRaw.zipWithIndex.foreach { case (entry, idx) =>
+      assert(columnDefs.contains(idx), s"column with index=$idx not found in column definitions derived from headerFile")
+      columnDefs(idx) match {
+        case ColumnDef(_, ColumnType.StartId, _) =>
+          startId = entry.toInt
+        case ColumnDef(_, ColumnType.EndId, _) =>
+          endId = entry.toInt
+        case ColumnDef(_, ColumnType.Type, _) =>
+          label = entry
+        case ColumnDef(Some(name), valueType, false) =>
+          parseProperty(entry, name, valueType).foreach(properties.addOne)
+        case ColumnDef(Some(name), valueType, true) =>
+          parseArrayProperty(entry, name, valueType).foreach(properties.addOne)
+        case other =>
+          throw new MatchError(s"unhandled case $other")
+      }
+    }
+    assert(startId != null, s"no START_ID column found in line $lineNo")
+    assert(endId != null, s"no END_ID column found in line $lineNo")
+    assert(label != null, s"no LABEL column found in line $lineNo")
+
+    val ret = ParsedEdgeRowData(startId, endId, label, properties.result())
+    logger.debug("parsed line {}: {}", lineNo, ret)
+    ret
+  }
+
+  private def parseProperty(rawValue: String, name: String, valueType: ColumnType.Value): Option[ParsedProperty] = {
+    if (rawValue != "" || valueType == ColumnType.String)
```

Comment:
Is `valueType == ColumnType.String` some temporary check? Asking because here we're very strict about `valueType`, basically filtering in only String, while `parsePropertyValue` actually handles much more types

Reply:
This is a bit confusing - I added a comment in b9546ec5f84c47ffe26e7870fc7b6b841c67ee67, hopefully that makes more sense?
```
    // an empty string is only a valid value if the valueType is String - otherwise it simply indicates an unset property
```

Reply:
Yes, it makes sense. Actually my initial conclusion came from me misreading the condition, I've taken `||` as `&&`.